### PR TITLE
Remove obsolete arcade-services submodule and upgrade to netcoreapp3.0 version of Darc.

### PR DESCRIPTION
**Pull request by crummel**
After https://github.com/dotnet/arcade-services/pull/778, we can upgrade to a new version of Darc that supports netcoreapp3.0, which should fix issues that we've seen on ARM and newer RIDs.  Also removing the obsolete submodule now that it's no longer used (fixes https://github.com/dotnet/source-build/issues/1354).

### CODE REVIEW

File: build-source-tarball.sh [hunk 1]
```diff
@@ -148,9 +148,6 @@ git submodule foreach --quiet --recursive '
 # Now re-uberclone into the tarball src directory.  Since we reuse the .gitdirs, this shouldn't hit the network at all.
 ignored_repos="https://dev.azure.com/dnceng/internal/_git/dotnet-optimization;https://dev.azure.com/devdiv/DevDiv/_git/DotNet-Trusted;https://devdiv.visualstudio.com/DevDiv/_git/DotNet-Trusted;https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-optimization;https://dev.azure.com/dnceng/internal/_git/dotnet-core-setup;https://github.com/dotnet/source-build-reference-packages"
 
-#export the LC_LIB_PATH for libgit2 so file as fedora fails to find it in the repodir
-export LD_LIBRARY_PATH=$CLI_PATH/tools/.store/microsoft.dotnet.darc/$DarcVersion/microsoft.dotnet.darc/$DarcVersion/tools/netcoreapp2.1/any/runtimes/rhel-x64/native/
```
Comment (adaggarwal) [inline build-source-tarball.sh]: We don't need this now? Is this due to netcoreapp3.0 update?

Comment (crummel) [reply build-source-tarball.sh]: I believe it's actually fixed by https://github.com/libgit2/libgit2sharp/pull/1714, but that change relied on some 3.0 runtime changes, so the fix didn't actually work until we switched to 3.0.

Comment (adaggarwal) [reply build-source-tarball.sh]: Thanks for the explanation. 👍

File: build-source-tarball.sh [hunk 2]
```diff
@@ -198,9 +195,7 @@ cp -r $SCRIPT_ROOT/scripts $TARBALL_ROOT/
 cp -r $SCRIPT_ROOT/repos $TARBALL_ROOT/
 cp -r $SCRIPT_ROOT/tools-local $TARBALL_ROOT/
 rm -rf $TARBALL_ROOT/tools-local/arcade-services/
```
Comment (omajid) [inline build-source-tarball.sh]: Can we remove this line too?

Comment (crummel) [reply build-source-tarball.sh]: Yes, thanks for the catch.

**Pull request merged**